public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,879 @@
+  These are different architectural paradigms, with their own strengths and weaknesses,
+  and you are free to choose between them.
+
+  The balance of this _reactive forms_ guide assumes the _reactive_ paradigm and 
+  concentrates exclusively on the reactive forms technology and techniques.
+
+  You'll learn about reactive forms by building one from scratch,
+  with instances of the Angular form classes.
+
+  ### Essential form classes
+
+  Before you begin, it may be helpful to read a brief description of the core form classes.
+  
+  * [_AbstractControl_](../api/forms/index/AbstractControl-class.html-class.html "API Reference: AbstractControl")
+  is the abstract base class for the three concrete form control classes: 
+  `FormControl`, `FormGroup`, and `FormArray`.
+  It provides their common behaviors and properties, some of which are _observable_.
+
+  * [_FormControl_](../api/forms/index/FormControl-class.html "API Reference: FormControl") 
+  tracks the value and validity status of an _individual_ form control.
+  It corresponds to an HTML form control such as an input box or selector.
+
+  * [_FormGroup_](../api/forms/index/FormGroup-class.html "API Reference: FormGroup")
+  tracks the value and validity state of a _group_ of `AbstractControl` instances.
+  The group's properties are its child controls.
+  The top-level form in your component is a `FormGroup`.
+
+  * [_FormArray_](../api/forms/index/FormArray-class.html "API Reference: FormArray")
+  tracks the value and validity state of a numerically indexed _array_ of `AbstractControl` instances.
+
+  You'll learn more about these classes as you build your first reactive form.
+
+.l-main-section
+a#setup
+:marked
+  ## Setup
+
+  Follow the steps in the [_Setup_ guide](../setup.html "Setup guide") 
+  for creating a new project folder (perhaps called `reactive-forms`)
+  based on the _QuickStart seed_.
+
+.l-main-section
+a#data-model
+:marked
+  ## Create a data model
+  The focus of this guide is a reactive forms component that edits a hero.
+  You'll need a `hero` class and some hero data.
+  Create a new `data-model.ts` file in the `app` directory and copy the content below into it.
+
++makeExample('reactive-forms/ts/app/data-model.ts', '','app/data-model.ts')(format=".")
+
+:marked
+  The file exports two classes and two constants. The `Address` 
+  and `Hero` classes define the application _data model_. 
+  The `heroes` and `states` constants supply the test data. 
+
+.l-main-section
+a#create-component
+:marked
+  ## Create a _reactive forms_ component
+  Make a new file called 
+  `hero-detail.component.ts` in the `app` directory and import these symbols:
+
++makeExample('reactive-forms/ts/app/hero-detail.component.ts', 'reactive-comp-imports','app/hero-detail.component.ts')(format=".")
+
+:marked
+  Here’s a summary description of the imported symbols, whose meaning and use
+  will become clear over the course of this guide:
```

Comment:
It is efficient providing all of these imports here at the beginning. But this may be misleading as the majority of these things are *not* required to successfully build a reactive form. It would be really nice to have a "first example" that showed the bare bones required to build a reactive form without the input/output /onchanges /eventemitter /FormBuilder /FormArray /etc. Some of these are needed only because of the way our example is set up. Others are only needed for more advanced scenarios.